novawallet/Common/Services/ExtrinsicService/Substrate/ExtrinsicSenderResolution/Delegation/DelegationResolutionPathFinder.swift

```diff
@@ -164,3 +179,38 @@ extension ProxyResolution {
         }
     }
 }
+
+extension DelegationResolution.PathFinder {
+    struct ProxyDelegationValue: AccountDelegationPathValue {
+        let proxyType: Proxy.ProxyType
+
+        func wrapCall(
+            _ call: JSON,
+            delegation: DelegationKey,
+            context: RuntimeJsonContext
+        ) throws -> JSON {
+            try Proxy.ProxyCall(
+                real: .accoundId(delegation.delegate),
+                forceProxyType: proxyType,
+                call: call
+            )
+            .runtimeCall()
+            .toScaleCompatibleJSON(with: context.toRawContext())
+        }
+    }
+
+    struct MultisigDelegationValue: AccountDelegationPathValue {
+        let threshold: UInt16
+        let signatories: [AccountId]
+
+        func wrapCall(
+            _ call: JSON,
+            delegation _: DelegationKey,
+            context _: RuntimeJsonContext
+        ) throws -> JSON {
+            // TODO: - Implement call wrapping
+
+            call
```

Comment:
So, here i would consider the following approaches (not sure yet which one to pick):

1) Check whether the call is from multisig pallet and intended to be signed by the same signatory as we have resolved. The idea here is to wrap only with as_multi but we need to differentiate from the cases where multisig is signatory of another multisig.

2) Introduce dummy call for approval and rejection and check for them here.

3) Wrap always here with as_multi. And when we perform an operation resolve the signatory in place and use current resolver for already wrapped call.

I would go with 3 for now as it one line for that function and continue the discussion in the approval/rejection operation task.